### ISSUE TITLE
lang/funcs: add "all" function 

### ISSUE THREAD

**Issue by artburkart** [hashicorp/terraform#25695]
### Current Terraform Version

```
0.12.29
```

### Use-cases

With the new validation functionality, it will be nice if we can validate that every object in a list type variable meet whatever expectations you have. For example, it would be nice if we could make the intent that a given key on every object has an appropriate value set on it.

```hcl
variable "amis" {
  type = list(object({
    id = string
  }))

  validation {
    condition = (all([
      for a in var.amis : length(a.id) > 4 && substr(a.id, 0, 4) == "ami-"
    ]))
    error_message = "The ID of at least one AMI was invalid."
  }
}
```

### Attempted Solutions

I'm not really sure how to solve this with Terraform's current functionality. Once I get a list of boolean values, I'm not sure what I can do to prove they're all truthy. I think the example I already provided above is sufficient to give a proper sense of the problem.

### Proposal

I've implemented an `all` collection function here: #25656
It checks whether all values in a list are either `true` or `"true"`.

### References

#25656

**Comment by alisdair** [comment]
> I'm not really sure how to solve this with Terraform's current functionality.

I just wanted to note that I think this is possible with existing functions and operators. For example, you could use `!contains`:

```hcl
  validation {
    condition = !contains([
      for a in var.amis : length(a.id) > 4 && substr(a.id, 0, 4) == "ami-"
    ], false)
    error_message = "The ID of at least one AMI was invalid."
  }
```

Alternatively, if you can validate each item in the list with a regex, wrap the result in `can`:

```hcl
  validation {
    condition = can([for a in var.amis : regex("^ami-", a.id)])
    error_message = "The ID of at least one AMI was invalid."
  }
```

**Comment by artburkart** [comment]
Oh interesting, that usage of `can` wasn't obvious to me before you shared your example. I kept putting it inside my list comprehension.

**Comment by artburkart** [comment]
I guess I have to bump this to make sure the bot doesn't come through and stale it.

**Comment by bflad** [comment]
I believe this is covered with the addition of the [`alltrue`](https://www.terraform.io/docs/language/functions/alltrue.html) and [`anytrue`](https://www.terraform.io/docs/language/functions/anytrue.html) collection functions available in Terraform 0.14 and later, but please reach out if that is not the case. 👍
